EIPS/eip-5792.md

```diff
@@ -1,390 +1,269 @@
-from the user.
-
-The current interface also does not work well for account abstracted wallets (e.g. [EIP-4337](./eip-4337.md)
-or [EIP-3074](./eip-3074.md)), which often involve a relaying service to sign and broadcast the transaction that
-triggers the function calls from the user's wallet. In these cases the actual transaction hash may not be known at the
-time of user signing, but must still be returned by `eth_sendTransaction`. The transaction hash returned
-by `eth_sendTransaction` in these cases is unlikely to be relevant to the transaction hash of the included transaction.
-The existing interface also provides no way to delay the resolution of the transaction hash, since it is used as the key
-of the transaction tracked by the dapp. Dapps often link to the block explorer for the returned transaction hash, but in
-these cases the transaction hash is wrong and the link will not work.
-
-Dapps need a better interface for sending batches of function calls from the user's wallet so they can interact with
-wallets without considering the differences between wallet implementations. These methods are backwards compatible with
-existing EOA wallets. EOA wallets may send bundles of function calls as individual transactions. The goal of the new
-interface is to be a more stable and flexible interface that enables a better user experience, while wallet
-implementations evolve over time.
+The current methods used to send transactions from the user wallet and check their status are `eth_sendTransaction` and `eth_getTransactionReceipt`.
+
+The current methods used to send transactions from the user wallet and check their status do not meet modern developer demands and cannot accomodate new transaction formats. Even the name–`eth_sendTransaction`–is an artifact of a time when nodes served as wallets. 
+
+Today, developers want to send batch transactions, which many smart accounts can support atomically. Developers also want to make use of features afforded by new transaction formats, like paymasters in [ERC-4337](./eip-4337.md) transactions. `eth_sendTransaction` offers no way to do these things. 
+
+In updating to a new set of `wallet_` RPCs, our main goals are to enforce a clean separation of wallet and app concerns, enable developers to make use of things like paymasters and batch transactions, and to create a clear way for more features to be added over time with minimal coordination.
 
 ## Specification
 
-Three new JSON-RPC methods are added. Dapps may begin using these methods immediately, falling back
-to `eth_sendTransaction` and `eth_getTransactionReceipt` when they are not available.
+Four new JSON-RPC methods are added. Apps may begin using these methods immediately, falling back to `eth_sendTransaction` and `eth_getTransactionReceipt` when they are not available.
+
+We define one capability related to call execution atomicity.
+
+### `wallet_getCapabilities`
```

Comment:
I suggest to remove this method and instead to reference CAIPs (which you can now reference from EIPs/ERCs the way I did when I [updated this same EIP a few months back](https://github.com/arein/dapp-smart-account/pull/1/files#diff-2ce6a27d54a5083f40751babba9d77c5529f0515cd5f8dc31a379406d52ff6cdR38).

This solution I argue is better as it doesn't ever require a roundtrip to discover the capabilities.

Reply:
What do you mean by “context switch”? 

`eth_getCapabilities` should mostly be a synchronous call just like `eth_accounts`. CAIP-25 is an orthogonal abstraction to this, that said,  consumers + wallets using CAIP-25 could include `eth_getCapabiltities` in their required/optional methods.

Reply:
What do you mean by “context switch”?

`eth_getCapabilities` should mostly be a synchronous call just like `eth_accounts`. CAIP-25 is an orthogonal abstraction to this, that said, consumers + wallets using CAIP-25 could include `eth_getCapabilities` in their required/optional methods.

Reply:
Anything capabilities or permissions related should not be on this EIP. The Permission Standard is still in discussion and this should wait for that to move forward, since this seems like it would be dependent on the outcome of the Permission Standard. This was originally an EIP for batched function calls, imho it should stay that way, unless you can find a way to remove the capabilities, for the time being. I think it would be best to wait for Permissions Standard to move forward first.

Reply:
@McOso what permissions standard are you referring to?

If the one Derek and others are working on, we became aware of that after we had already published our plans here. I think we have different goals and designs, and I don't think it's fair to ask us to wait on something someone else is working on which is independent of what we're doing :)  

If we can find a common best path that's great. But doesn't seem to be the case so far.

If you are sincerely concerned on the change to this EIP, the alternative path was a new proposal with these same RPCs. We took over this EIP because of the similarity in the RPCs and Moody asked us to. Otherwise this EIP is stagnant and nothing is going to happen with it. 

We should not solve for batch transactions as a one off. We need to fix several things here.

Reply:
The outcome of any "permission standard" EIP would fit into this type of method, no?

IIRC, the goal of the "permission standard" is to define a standard for a consumer to **asynchronously request permissions**,
Whereas, the goal of this method is to define a standard for a consumer to **synchronously discover capabilities**.

If there is a way to synchronously discover capabilities in the permissions standard discussions, then I am all ears and maybe these efforts should be combined!

Reply:
This standard needs no mention of the WIP session key/permissions standard @McOso is referring to.

For session keys the dApp will send the UserOps/Transactions. They can still use these new RPC methods to do so.

In fact I see these two things in harmony.

The [UserOpBuilder ERC](https://zerodev.notion.site/ERC_UserOpBuilder-753e73db33bd4c8c8a9fbeeaf160a27c?pvs=4) may deserve a mention here. But since it's just going into draft this week it's probably better not to.